torch/distributed/checkpoint/state_dict_saver.py

```diff
@@ -256,6 +302,42 @@ def async_save(
     return f
 
 
+def _async_save_process_based(
+    state_dict: STATE_DICT_TYPE,
+    checkpoint_id: Union[str, os.PathLike, None],
+    storage_writer: StorageWriter,
+    planner: Optional[SavePlanner] = None,
+    process_group: Optional[dist.ProcessGroup] = None,
+) -> Future:
+    """
+    Asynchronous version of ``save``. This code first de-stages the state_dict on to the
+    staging storage (defaults to SHARED-MEMORY), and then calls the `save` in a separate PROCESS.
+    """
+
+    local_sync_state: Optional[LocalSyncState] = None
+    if isinstance(storage_writer, AsyncProcessStager):
+        staged_state_dict, local_sync_state = storage_writer.stage(state_dict)
```

Comment:
I was messing around with this code in a local environment and run into an error with this line. The default `storage_writer.stage(state_dict)` for `async_save` returns 1 object, not 2

Reply:
I no longer run into this issue with the latest changes.